### ISSUE TITLE
Consider switching to a feature branch workflow

### ISSUE THREAD

**Issue by dewydex** [OpenVPN/easy-rsa#615]
**TL;DR it might be a good idea to start developing in feature branches and merging into master...**

It's excellent to see so much active development on Easy-RSA. What may not be adequately reflected here at GitHub is the degree to which this software pervades in the wild, especially in offline or decentralized forms.

It's something people have been hammering on, with, and around for a long time ; or alternatively assume to be dirt-simple and rock-solid. Much of this spirit and nature come from its official distribution along side tools like "OpenVPN".

Through that lens I'd like to express the possibility that so much active development being poured directly into `master`, even as it's presently considered the "development" branch, presents a risk to how new versions of this software are going to find their place alongside its earlier iterations.

To restate: the project is approaching the concern that what version you're running-or-downloading is going to matter far more than it did previously, that's a bit counter-intuitive to its "Easy" namesake, and it's unclear presently if parent communities (like "OpenVPN") are well prepared to handle the tool evolving in the present workflow.

Switching to "feature branch" development, and merging those features into the master, might provide some added clarity of thought and purpose, and in that way help this software to grow in a way that will trend towards stability.

Here are some reference materials on the subject for those unfamiliar...
* https://docs.github.com/en/get-started/quickstart/github-flow
* https://docs.gitlab.com/ee/gitlab-basics/feature_branch_workflow.html
* https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow
* https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging

**Comment by ecrist** [comment]
Given other challenges, and that we are recently adopting our current workflow, I don't support another change at this time.

Eric Crist

On Jun 24, 2022, at 1:39 PM, Roughstone ***@***.***> wrote:

﻿

TL;DR it might be a good idea to start developing in feature branches and merging into master...

It's excellent to see so much active development on Easy-RSA. What may not be adequately reflected here at GitHub is the degree to which this software pervades in the wild, especially in offline or decentralized forms.

It's something people have been hammering on, with, and around for a long time ; or alternatively assume to be dirt-simple and rock-solid. Much of this spirit and nature come from its official distribution along side tools like "OpenVPN".

Through that lens I'd like to express the possibility that so much active development being poured directly into master, even as it's presently considered the "development" branch, presents a risk to how new versions of this software are going to find their place alongside its earlier iterations.

To restate: the project is approaching the concern that what version you're running-or-downloading is going to matter far more than it did previously, that's a bit counter-intuitive to its "Easy" namesake, and it's unclear presently if parent communities (like "OpenVPN") are well prepared to handle the tool evolving in the present workflow.

Switching to "feature branch" development, and merging those features into the master, might provide some added clarity of thought and purpose, and in that way help this software to grow in a way that will trend towards stability.

Here are some reference materials on the subject for those unfamiliar...

  *   https://docs.github.com/en/get-started/quickstart/github-flow
  *   https://docs.gitlab.com/ee/gitlab-basics/feature_branch_workflow.html
  *   https://www.atlassian.com/git/tutorials/comparing-workflows/feature-branch-workflow
  *   https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging

**Comment by TinCanTech** [comment]
FTR; I would not be opposed to `a.n.other`  branch.

**Comment by ecrist** [comment]
I'll make a 3.2.x branch tomorrow.

Eric Crist

On Jun 25, 2022, at 5:46 PM, TinCanTech ***@***.***> wrote:

﻿

FTR; I would not be opposed to a.n.other branch.
